src/clojure/puppetlabs/ssl_utils/core.clj

```diff
@@ -873,6 +873,33 @@
     (SSLUtils/caCertAndCrlPemsToSSLContext ca-cert-reader
                                                        crls-reader)))
 
+(defn generate-ssl-context
+  [options]
```

Comment:
@nwolfe The idea for this function came from a PR I opened last week where I had to configure SSL. @rlinehan noticed that I was extracting SSL Options from a map myself and turning them into an SSLContext, and that a number of clojure projects have had to do the same thing, so we decided it would be good to standardize this in a single function in `jvm-ssl-utils`.

I've added a docstring to the clojure function, I'll go ahead and add a Java method with docs and modify the clojure function to be a wrapper around the Java method.